server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader9to10.java

```diff
@@ -111,6 +111,31 @@
  * {@link #dropSortedMapWALFiles(ServerContext)}. For more information see the following issues:
  * <a href="https://github.com/apache/accumulo/issues/2117">#2117</a> and
  * <a href="https://github.com/apache/accumulo/issues/2179">#2179</a>
+ *
+ * The following methods were created for External compactions:
+ * {@link #createExternalCompactionNodes(ServerContext)}, {@link #setMetaTableProps(ServerContext)}
```

Comment:
A lot of this is very verbose and could be eliminated to streamlined. Some of the verbosity that explains how things changed over time can be removed. The javadoc should document the current code, not the previous code or the changes. Changes are documented in the change log / git history. In this case, the above comment is explaining the reason why these methods were added, but the purpose of those methods should be in the javadoc for those specific methods. They don't need to be here at the class level.

Reply:
I disagree. The upgrade code is specifically about past changes to code and is not clear what methods are doing what. But this documentation was more meant to be a bridge for to assist admins performing upgrades. It is a lot easier for an admin to find the line in the code an exception refers to (something they can even do in the browser) then to track git history. Do you have any specific suggestions?